### DISCUSSION TITLE
PowerDNS Authoritative Server - CATALOG-HASH is recalculated every minute

### DISCUSSION THREAD

**Discussion by velechac** [PowerDNS/pdns#13856] [Q&A] [answered]
Hello,
We have an authoritative PowerDNS server with two catalog zones.
All zones are members of one of these catalog zones.

Although there are no updates to the member zones or their records, a new CATALOG-HASH is recalculated every minute. This increments the SOA serial catalog zone and transfers it (every minute).

The tested version is pdns-server 4.8 and 4.9 in combination with the PostgreSQL backend. We tried to re-run `rectify-all-zones`, but there was no change.

How can we get CATALOG-HASH to recalculate only when members change?

Regards

**Comment by Habbie** [comment]
Do you have SOA-EDIT set on any of your member zones?

**Comment by velechac** [comment]
There is no SOA-EDIT in domainmetadata. We have only global configuration:
```
default-soa-edit=NONE
default-soa-edit-signed=INCEPTION-EPOCH
```

**Comment by marekj11** [comment, marked as answer]
Hello,

we checked the query `info-all-master/primary-query`.
By default, PostgreSQL returns data in an unspecified order.
The query outputs need to be sorted for the hash algorithm for an unchanged hash if the domains data has not changed.

[PostgreSQL queries order docs](https://www.postgresql.org/docs/current/queries-order.html)
```
If sorting is not chosen, the rows will be returned in an unspecified order. The actual order in that case will depend on the scan and join plan types and the order on disk, but it must not be relied on. 
```

Fixed query:
```
info-all-master/primary-query: select domains.id, domains.name, domains.type, domains.notified_serial, domains.options, domains.catalog, records.content from records join domains on records.domain_id=domains.id and records.name=domains.name where records.type='SOA' and records.disabled=false and domains.type in ('MASTER', 'PRODUCER') ORDER BY domains.id
```

Regards

**Reply by Habbie** [reply]
thank you for finding that. I have put it in #14454
